app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,1234 @@
+                    message: customError
+                }
+            };
+        }
+        if employee is () {
+            log:printDebug(string `${types:NOT_FOUND_CUSTOM_ERROR}: ${email}`);
+            return <http:NotFound>{
+                body: {
+                    message: string `${types:NOT_FOUND_CUSTOM_ERROR}: ${email}`
+                }
+            };
+        }
+
+        return employee;
+    }
+
+    # Get basic information of employees.
+    #
+    # + filters - Filter objects containing the filter criteria for the query
+    # + 'limit - The maximum number of employees to return
+    # + offset - The number of employees to skip before starting to collect the result set
+    # + return - Basic information of the employees or an error
+    resource function post employees(types:EmployeeFilter? filters, int? 'limit, int? offset)
+        returns types:Employee[]|http:BadRequest|http:InternalServerError|http:NotFound {
+
+        types:Employee[]|error employees =
+            database:getEmployees(filters ?: {}, 'limit ?: types:DEFAULT_LIMIT, offset ?: 0);
+        if employees is error {
+            string customError = "Error getting employee information!";
+            log:printError(customError, employees);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+        return employees;
+    }
+
+    # Organization structure data retrieval.
+    #
+    # + return - Internal Server Error or Organization structure data
+    resource function get org\-structure() returns types:OrgStructure|http:InternalServerError {
+
+        if orgStructureCache.hasKey(types:ORG_STRUCTURE_CACHE_KEY) {
+            any|error cacheOrg = orgStructureCache.get(types:ORG_STRUCTURE_CACHE_KEY);
+            if cacheOrg is types:OrgStructure {
+                return cacheOrg;
+            }
+            if cacheOrg is cache:Error {
+                string customError = "Error retrieving employee from cache";
+                log:printError(customError, cacheOrg);
+            }
+        }
+        types:BusinessUnit[]|error businessUnits = database:getOrgDetails(
+                filter = {},
+                'limit = types:DEFAULT_LIMIT,
+                offset = 0
+            );
+        if businessUnits is error {
+            string customError = "Error getting business units";
+            log:printError(customError, businessUnits);
+            return <http:InternalServerError>{
+                body: {
+                    message: customError
+                }
+            };
+        }
+
+        types:Company[]|error companies = database:getCompanies(
+                filter = {},
+                'limit = types:DEFAULT_LIMIT,
+                offset = 0
+            );
+        if companies is error {
+            string customError = "Error getting companies";
+            log:printError(customError, companies);
```

Comment:
We are logging the same error in multiple places. e.g.: inside the `getCompanies` function, and here, you logged the error with stack trace. It would be better, if we transferred the original error with proper message and logged it in here.

Reply:
ACK. We've discussed this internally as well. Thanks for the suggestion.